src/Clip.cpp

```diff
@@ -1269,7 +1300,7 @@ void Clip::apply_keyframes(std::shared_ptr<Frame> frame, std::shared_ptr<QImage>
     painter.setTransform(transform);
 
     // Composite a new layer onto the image
-    painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
+    painter.setCompositionMode((QPainter::CompositionMode) blend);
```

Comment:
So, here's where you'd need to make the last change. Instead of passing the value of `blend` to the QPainter instance directly, you'll need to map it from our enum to theirs.

Probably easiest to write a small utility method for this, but ultimately up to you. But assuming you do write one, let's call it
```c++
QPainter::CompositionMode Clip::blendToQPainter(openshot::BlendMode blend) {
    // ...
}
```
There's a bunch of different ways you can do the mapping (including building a `std::map<openshot::BlendMode, QPainter::CompositionMode>` to index the Qt modes by our enum), but there are some advantages to sticking with the basics.

If you use a basic `switch(blend)` to drive the conversion, and you _don't_ give it a `default:` branch, then the compiler will (with the right settings) emit a compile-time warning if any of the `BlendMode` enum values are missed by the `switch` statement. (So, you _should_ include a case for `BLEND_SOURCE_OVER` explicitly, even though it's the default. Just to silence those warnings.)

The compiler's monitoring of `switch` usage becomes especially handy further down the road, when someone decides to change or add elements to the `enum`, and forgets to update the code that uses it. 

But basically that's it, just fill it with a bunch of cases like
```c++
case BLEND_DESTINATION_ATOP:
    painter_mode = QPainter::CompositionMode_DestinationAtop;
    break;
case BLEND_HARD_LIGHT:
    painter_mode = QPainter::CompositionMode_HardLight;
    break;
```
Lather rinse, repeat, and finally `return painter_mode;`. The return value of `blendToQPainter()` you could even pass directly to `painter.setCompositionMode()` — you don't really need it for anything else. But that'a entirely up to you.